cmd/snap-discard-ns/snap-discard-ns.c

```diff
@@ -56,7 +65,12 @@ int main(int argc, char** argv) {
      * during normal operation but it is not preserved across the life-cycle of
      * the process anyway so no attempt is made to unlock it ahead of any call
      * to die() */
-    int snap_lock_fd = sc_lock_snap(snap_instance_name);
+    int snap_lock_fd = -1;
```

Comment:
Won't flock() in sc_unlock() fail with EBADF now? Please double check if you need to `if (sc_snap_lock_fd >= 0) { .. }` that code.